src/main/java/de/fraunhofer/aisec/cpg/frontends/java/JavaLanguageFrontend.java

```diff
@@ -119,25 +119,36 @@ public TranslationUnitDeclaration parse(File file) throws TranslationException {
       TranslationUnitDeclaration fileDeclaration =
           NodeBuilder.newTranslationUnitDeclaration(file.toString(), context.toString());
       setCurrentTU(fileDeclaration);
-      TranslationUnitDeclaration declaration = fileDeclaration;
+      Declaration declaration = fileDeclaration;
 
       PackageDeclaration packDecl = context.getPackageDeclaration().orElse(null);
       NamespaceDeclaration namespaceDeclaration = null;
       if (packDecl != null) {
         namespaceDeclaration = NodeBuilder.newNamespaceDeclaration(packDecl.getName().asString());
         // Todo set region and code and push/pop scope
         scopeManager.enterScope(namespaceDeclaration);
-        declaration.add(namespaceDeclaration);
+        if (declaration instanceof TranslationUnitDeclaration) {
+          ((TranslationUnitDeclaration) declaration).add(namespaceDeclaration);
+        }
         declaration = namespaceDeclaration;
       }
 
       for (TypeDeclaration<?> type : context.getTypes()) {
-        declaration.add(getDeclarationHandler().handle(type));
+        Declaration typeD = getDeclarationHandler().handle(type);
+        if (declaration instanceof TranslationUnitDeclaration) {
+          ((TranslationUnitDeclaration) declaration).add(typeD);
+        } else {
+          scopeManager.addDeclaration(typeD);
```

Comment:
Not sure I understand this change (and the ones below and above). In which cases is it not a Translation Unit ?! Line 119 specifically creates a translation unit decl.

Reply:
We would not have to make a difference here if there was a scope for `TranslationUnitDeclaration`, which may actually be something that we can introduce in the future. Previously a NamespaceDeclaration extended from a TranslationUnit, with this PR NamespaceDeclarations are no longer TranslationUnitDeclarations, which IMO they shouldn't be. In java, for a long time now we had a `TranslationUnitDeclaration` containing a `NamespaceDeclaration` which then contains the rest of the file. This served the purpose of unifying how packages and namespaces work.

So basically this is for the case that the condition that there is a `packageDecl` and the statement in line 133 is executed `declaration = namespaceDeclaration;`.